### PR TITLE
[tests-only] Harden smoke tests

### PR DESCRIPTION
**Pull request by fschade**
## Description
as title says, the smoke tests also started to get flaky in the last weeks, this is a testing pr to see if and how we can fix knowing cases. On top of that it also adds a few developer goodies

Following things are included:
* add cli option to record hars
* add cli option to record a video
* add cli option to record traces
* add cucumber reportings
* add cli option to disable downloads
* prefer locators api over others
* force record traces, hars and videos on retry
* refactor out hard cucumber coupling for later (maybe integration tests!?)
* rename smoke tests to e2e since this becomes the standard
* add logger and log levels to get more context while developing

the list of this pr's improvements can grow over time but doesn't have to if we reach a point where its ok.

## Related Issue

* https://github.com/owncloud/web/issues/6163
* https://github.com/owncloud/web/issues/6052
* https://github.com/owncloud/web/issues/6051
* https://github.com/owncloud/web/pull/6077
* https://github.com/owncloud/web/pull/6154
* https://github.com/owncloud/web/pull/6149
* https://github.com/owncloud/web/pull/6143

## Motivation and Context
trust our tests

### CODE REVIEW

File: tests/smoke/support/world/actor.ts
Comment (ScharfViktor) [inline tests/smoke/support/world/actor.ts:29]: Mayby use another date format? for example: `('YYYY/MM/D hh:mm:ss')`

**Review by ScharfViktor — APPROVED** [review]
What you did is very amazing. I love playwright
- we can get rid of `waitForSelector`
- I think it would be cool to add a description "how create tracing and show log" to Readmi (this can also be done later) and try to create trace in CI `npx playwright show-trace https://example.com/trace.zip`

File: tests/smoke/support/cta/files/sidebar.ts
Comment (fschade) [reply tests/smoke/support/cta/files/sidebar.ts:26]: Good idea ❤️

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingIntGroupsToRoot</strong> https://drone.owncloud.com/owncloud/web/21177/29/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by fschade** [comment]
@phil-davis, @ScharfViktor  were now able to record reports, videos, logs, hars and traces. Can you please have a look how we can transport them to github in error case!?

it's ok to not have it in this pr, i would prefer to get it to master asap even if we still have 2 flaky sports in the tests. Then we can build something on top of it.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingIntGroupsToRoot</strong> https://drone.owncloud.com/owncloud/web/21178/29/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingIntGroups</strong> https://drone.owncloud.com/owncloud/web/21180/28/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingIntGroupsToRoot</strong> https://drone.owncloud.com/owncloud/web/21181/29/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Review by kulmann — APPROVED** [review]
Awesome! Especially the traces will help a lot...  nice work!
Added some comments, I'm sure you'll take care of them as needed. :-)

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingIntGroupsToRoot</strong> https://drone.owncloud.com/owncloud/web/21192/29/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by phil-davis** [comment]
PR #6170 was merged to fix more nightwatch expiration-date test problems.
I rebased this PR so that it gets up-to-date CI, hopefully with unrelated nightwatch test fails. Let's CI what CI  thinks.

**Review by phil-davis — COMMENTED** [review]
Looks OK to me. There are just a few comments from @kulmann then I suggesting squash the commits and merge.

**Comment by fschade** [comment]
Guys you’re awesome ❤️

File: .drone.star
Comment (pascalwengerter) [inline .drone.star:1103]: Why the `cucumber` part in the yarn cmd? I'd like it to be short&easy to type 😬

File: cucumber.js
Comment (pascalwengerter) [reply cucumber.js:12]: +1 :)

**Review by pascalwengerter — COMMENTED** [review]
Please don't merge before #6039 got merged, this one already is not rebase-able anymore and touches quite some stuff you're editing here, too

**Comment by phil-davis** [comment]
@fschade other sidebar etc. stuff was merged. IMO  you can rebase this and resolve conflicts.

File: tests/e2e/config.js
Comment (fschade) [reply tests/e2e/config.js:1]: we now consume config values in cucumber.js too, as this is POJS it can't access values from a ts file

File: cucumber.js
Comment (fschade) [reply cucumber.js:12]: i want to collect metrics over time to see which parts are flaky, recording videos, traces... etc. is a heavy task which slows down things. we now automatically torn this on in case of a failure: https://github.com/owncloud/web/blob/01b477166571fb633ff543f8618ab77caaa44882/tests/e2e/cucumber/environment/index.ts#L96 which then needs at least one retry

File: tests/e2e/README.md
Comment (fschade) [reply tests/e2e/README.md:50]: cc.: @ScharfViktor if i remember right you already started to write something!?

File: .drone.star
Comment (fschade) [reply .drone.star:1103]: cucumber is the bdd framework which drives our playwright tests, the tests are implemented abstract.
This means were able to have multiple e2e flavoured tests. For example one which uses playwright testing library. 

The idea is to have different wrappers for different needs, gherkin is nice for journeys written by product owner, but i expect that we try out to have also some integration like tests in place sooner or later. this is why we mention cucumber in the script

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingAccept</strong> https://drone.owncloud.com/owncloud/web/21338/15/1
The following scenarios passed on retry:
- webUISharingAcceptSharesToRoot/acceptShares.feature:108

File: tests/e2e/README.md
Comment (fschade) [reply tests/e2e/README.md:50]: @kulmann issue created, i close this here for now.
https://github.com/owncloud/web/issues/6191

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingAccept</strong> https://drone.owncloud.com/owncloud/web/21376/15/1
The following scenarios passed on retry:
- webUISharingAcceptSharesToRoot/acceptShares.feature:108

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPermission</strong> https://drone.owncloud.com/owncloud/web/21376/35/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingAndTrashbin</strong> https://drone.owncloud.com/owncloud/web/21376/34/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingSharingInd</strong> https://drone.owncloud.com/owncloud/web/21376/33/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingIntUsers2</strong> https://drone.owncloud.com/owncloud/web/21376/31/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingIntGroupsToRoot</strong> https://drone.owncloud.com/owncloud/web/21376/29/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oc10SharingIntUsers3</strong> https://drone.owncloud.com/owncloud/web/21376/32/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingExternal</strong> https://drone.owncloud.com/owncloud/web/21377/40/1
The following scenarios passed on retry:
- webUISharingExternal/federationSharing.feature:106

**Pull request merged**
